manifests/osfamily/solaris.pp

```diff
@@ -32,6 +32,57 @@
         source => "puppet:///pe_packages/${pe_server_version}/${::platform_tag}/solaris-noask",
       }
     }
+    '11': {
+      class { 'puppet_agent::prepare::package':
+        package_file_name => $package_file_name,
+      }
+      contain puppet_agent::prepare::package
+
+      # Backup user configuration because solaris 11 will blow away
+      # /etc/puppetlabs/ when uninstalling the pe-* modules.
+      file { '/tmp/puppet_agent/':
+        ensure => directory,
+      }
+      exec { 'puppet_agent backup /etc/puppetlabs/':
```

Comment:
Yes; looks like you already added a test, or I was blind when I wrote the comment.